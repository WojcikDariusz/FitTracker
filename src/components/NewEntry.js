import React from 'react';

class NewEntry extends React.Component {

    constructor() {
        super();
        this.state = {
            activity : {
                type : "",
                distance : "",
                ifFinished : ""
            }
        }
   
    }
    //Due to arrow function we dont need to bind this.changeDistance= this.changeDistance.bind(this);
    changeValue = (event) => {
        
        let newValue = {
            [event.target.name] : event.target.value
        }
        
        this.setState({
            activity : {
                [event.target.name] : newValue
            }
        });
    
    }


    render() {

        

        return (
            <div className='newEntry col-xs-4'>
                <form>
                <div className="form-group">
                    <select multiple className="form-control" id="activityType" name="type" onChange={this.changeValue}>
                        <option name="running">Running</option>
                        <option name="walking">Walking</option>
                        <option name="swimming">Swimming</option>
                        <option name="rollerSkating">Roller skating</option>
                    </select>
                </div>      
                <div className="form-group">      
                        <input type="number" placeholder="Distance [m]" id="distance" name="distance" className="form-control" onChange={this.changeValue} ></input>
                </div>
                <div className="form-group">
                    <input type="checkbox" id="ifFinished" name="ifFinished" onChange={this.changeValue}></input>
                    <label htmlFor="ifFinished"> Did you finished?</label>
                </div>
                <div className="form-group">
                        <input type="submit" value="Dodaj" id="submit" className="btn btn-primary"></input>
                </div>
                </form>
            </div>
        )
    };
};

export default NewEntry;