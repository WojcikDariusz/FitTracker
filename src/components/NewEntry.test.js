import React from 'react';
import ReactDOM from 'react-dom';
import NewEntry from './NewEntry';
import {configure,shallow} from 'enzyme';
import adapter from 'enzyme-adapter-react-16';

configure({adapter: new adapter()});

describe('New Entry tests', () => {

    it('Correctly renders the header html', () => {
        const div = document.createElement('div');
        ReactDOM.render(<NewEntry />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the text', () => {
        const wrapper = shallow(<NewEntry />);
        expect(wrapper.find('p').text()).toBe('NewEntry');
    });

});