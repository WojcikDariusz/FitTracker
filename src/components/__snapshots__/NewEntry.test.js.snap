// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Entry tests renders the text 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewEntry />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <div
          className="form-group"
        >
          <select
            className="form-control"
            id="activityType"
            multiple={true}
            name="activityType"
          >
            <option
              name="running"
            >
              Running
            </option>
            <option
              name="walking"
            >
              Walking
            </option>
            <option
              name="swimming"
            >
              Swimming
            </option>
            <option
              name="rollerSkating"
            >
              Roller skating
            </option>
          </select>
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="distance"
            name="distance"
            placeholder="Distance [m]"
            type="number"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            id="ifFinished"
            name="ifFinished"
            type="checkbox"
          />
          <label
            htmlFor="ifFinished"
          >
             Did you finished?
          </label>
        </div>
        <div
          className="form-group"
        >
          <input
            className="btn btn-primary"
            id="submit"
            type="submit"
            value="Dodaj"
          />
        </div>
      </form>,
      "className": "newEntry col-xs-4",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="form-group"
          >
            <select
              className="form-control"
              id="activityType"
              multiple={true}
              name="activityType"
            >
              <option
                name="running"
              >
                Running
              </option>
              <option
                name="walking"
              >
                Walking
              </option>
              <option
                name="swimming"
              >
                Swimming
              </option>
              <option
                name="rollerSkating"
              >
                Roller skating
              </option>
            </select>
          </div>,
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="distance"
              name="distance"
              placeholder="Distance [m]"
              type="number"
            />
          </div>,
          <div
            className="form-group"
          >
            <input
              id="ifFinished"
              name="ifFinished"
              type="checkbox"
            />
            <label
              htmlFor="ifFinished"
            >
               Did you finished?
            </label>
          </div>,
          <div
            className="form-group"
          >
            <input
              className="btn btn-primary"
              id="submit"
              type="submit"
              value="Dodaj"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <select
              className="form-control"
              id="activityType"
              multiple={true}
              name="activityType"
            >
              <option
                name="running"
              >
                Running
              </option>
              <option
                name="walking"
              >
                Walking
              </option>
              <option
                name="swimming"
              >
                Swimming
              </option>
              <option
                name="rollerSkating"
              >
                Roller skating
              </option>
            </select>,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  name="running"
                >
                  Running
                </option>,
                <option
                  name="walking"
                >
                  Walking
                </option>,
                <option
                  name="swimming"
                >
                  Swimming
                </option>,
                <option
                  name="rollerSkating"
                >
                  Roller skating
                </option>,
              ],
              "className": "form-control",
              "id": "activityType",
              "multiple": true,
              "name": "activityType",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Running",
                  "name": "running",
                },
                "ref": null,
                "rendered": "Running",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Walking",
                  "name": "walking",
                },
                "ref": null,
                "rendered": "Walking",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Swimming",
                  "name": "swimming",
                },
                "ref": null,
                "rendered": "Swimming",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Roller skating",
                  "name": "rollerSkating",
                },
                "ref": null,
                "rendered": "Roller skating",
                "type": "option",
              },
            ],
            "type": "select",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control"
              id="distance"
              name="distance"
              placeholder="Distance [m]"
              type="number"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "id": "distance",
              "name": "distance",
              "placeholder": "Distance [m]",
              "type": "number",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                id="ifFinished"
                name="ifFinished"
                type="checkbox"
              />,
              <label
                htmlFor="ifFinished"
              >
                 Did you finished?
              </label>,
            ],
            "className": "form-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "ifFinished",
                "name": "ifFinished",
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Did you finished?",
                "htmlFor": "ifFinished",
              },
              "ref": null,
              "rendered": " Did you finished?",
              "type": "label",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="btn btn-primary"
              id="submit"
              type="submit"
              value="Dodaj"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "btn btn-primary",
              "id": "submit",
              "type": "submit",
              "value": "Dodaj",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <div
            className="form-group"
          >
            <select
              className="form-control"
              id="activityType"
              multiple={true}
              name="activityType"
            >
              <option
                name="running"
              >
                Running
              </option>
              <option
                name="walking"
              >
                Walking
              </option>
              <option
                name="swimming"
              >
                Swimming
              </option>
              <option
                name="rollerSkating"
              >
                Roller skating
              </option>
            </select>
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="distance"
              name="distance"
              placeholder="Distance [m]"
              type="number"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              id="ifFinished"
              name="ifFinished"
              type="checkbox"
            />
            <label
              htmlFor="ifFinished"
            >
               Did you finished?
            </label>
          </div>
          <div
            className="form-group"
          >
            <input
              className="btn btn-primary"
              id="submit"
              type="submit"
              value="Dodaj"
            />
          </div>
        </form>,
        "className": "newEntry col-xs-4",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="form-group"
            >
              <select
                className="form-control"
                id="activityType"
                multiple={true}
                name="activityType"
              >
                <option
                  name="running"
                >
                  Running
                </option>
                <option
                  name="walking"
                >
                  Walking
                </option>
                <option
                  name="swimming"
                >
                  Swimming
                </option>
                <option
                  name="rollerSkating"
                >
                  Roller skating
                </option>
              </select>
            </div>,
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="distance"
                name="distance"
                placeholder="Distance [m]"
                type="number"
              />
            </div>,
            <div
              className="form-group"
            >
              <input
                id="ifFinished"
                name="ifFinished"
                type="checkbox"
              />
              <label
                htmlFor="ifFinished"
              >
                 Did you finished?
              </label>
            </div>,
            <div
              className="form-group"
            >
              <input
                className="btn btn-primary"
                id="submit"
                type="submit"
                value="Dodaj"
              />
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <select
                className="form-control"
                id="activityType"
                multiple={true}
                name="activityType"
              >
                <option
                  name="running"
                >
                  Running
                </option>
                <option
                  name="walking"
                >
                  Walking
                </option>
                <option
                  name="swimming"
                >
                  Swimming
                </option>
                <option
                  name="rollerSkating"
                >
                  Roller skating
                </option>
              </select>,
              "className": "form-group",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    name="running"
                  >
                    Running
                  </option>,
                  <option
                    name="walking"
                  >
                    Walking
                  </option>,
                  <option
                    name="swimming"
                  >
                    Swimming
                  </option>,
                  <option
                    name="rollerSkating"
                  >
                    Roller skating
                  </option>,
                ],
                "className": "form-control",
                "id": "activityType",
                "multiple": true,
                "name": "activityType",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Running",
                    "name": "running",
                  },
                  "ref": null,
                  "rendered": "Running",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Walking",
                    "name": "walking",
                  },
                  "ref": null,
                  "rendered": "Walking",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Swimming",
                    "name": "swimming",
                  },
                  "ref": null,
                  "rendered": "Swimming",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Roller skating",
                    "name": "rollerSkating",
                  },
                  "ref": null,
                  "rendered": "Roller skating",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="form-control"
                id="distance"
                name="distance"
                placeholder="Distance [m]"
                type="number"
              />,
              "className": "form-group",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "id": "distance",
                "name": "distance",
                "placeholder": "Distance [m]",
                "type": "number",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  id="ifFinished"
                  name="ifFinished"
                  type="checkbox"
                />,
                <label
                  htmlFor="ifFinished"
                >
                   Did you finished?
                </label>,
              ],
              "className": "form-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "ifFinished",
                  "name": "ifFinished",
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Did you finished?",
                  "htmlFor": "ifFinished",
                },
                "ref": null,
                "rendered": " Did you finished?",
                "type": "label",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="btn btn-primary"
                id="submit"
                type="submit"
                value="Dodaj"
              />,
              "className": "form-group",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "btn btn-primary",
                "id": "submit",
                "type": "submit",
                "value": "Dodaj",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
